pkg/kv/kvserver/store.go

```diff
@@ -1897,6 +1897,52 @@ func ReadStoreIdent(ctx context.Context, eng storage.Engine) (roachpb.StoreIdent
 	return ident, err
 }
 
+// loadAndReconcilesReplicas loads the Replicas present on this
+// store. It reconciles inconsistent state and runs validation checks.
+//
+// TODO(sep-raft-log): also load *uninitialized* Replicas.
+func loadAndReconcileReplicas(
+	ctx context.Context, eng storage.Engine,
+) (map[storage.FullReplicaID]*roachpb.RangeDescriptor, error) {
```

Comment:
Could you make this iterative instead? Don't much like this newly added temporary map which may be 100k+ on large clusters.

Reply:
Do you mind revisiting this later? Changing this here leads to a need to change in [`863932d` (#93317)](https://github.com/cockroachdb/cockroach/pull/93317/commits/863932d1d8ce1504a55ae9d39b37c048a6f4583f) and I'm not convinced spending this time is worth it.

Reply:
Yep we can revisit later! I understand that the change could've been made in the follow-up PR, not saying it would have been prohibitive. But it feels inefficient to make toss-up changes that ripple through a stack of PRs when there isn't a clear indication that they're better than the status quo. I'll tack a TODO on to the next stack so that we don't have to re-run CI here.